### ISSUE TITLE
knex-cli prints usage help text twice

### ISSUE THREAD

**Issue by emilwihlander** [knex/knex#4072]
# Environment

Knex CLI version: 0.21.6
Knex Local version: 0.21.2
Database + version: N/A
OS: Windows 10

# Bug

When writing `knex` in Powershell it prints the usage help text twice, it should only be printed once.

```
PS C:\***> knex
Usage:  [options] [command]

Options:
  -V, --version                   output the version number
  --debug                         Run with debugging.
  --knexfile [path]               Specify the knexfile path.
  --knexpath [path]               Specify the path to knex instance.
  --cwd [path]                    Specify the working directory.
  --client [name]                 Set DB client without a knexfile.
  --connection [address]          Set DB connection without a knexfile.
  --migrations-directory [path]   Set migrations directory without a knexfile.
  --migrations-table-name [path]  Set migrations table name without a knexfile.
  --env [name]                    environment, default: process.env.NODE_ENV || development
  --esm                           Enable ESM interop.
  --specific [path]               Specify one seed file to execute.
  --timestamp-filename-prefix     Enable a timestamp prefix on name of generated seed files.
  -h, --help                      display help for command

Commands:
  init [options]                          Create a fresh knexfile.
  migrate:make [options] <name>           Create a named migration file.
  migrate:latest [options]                Run all migrations that have not yet been run.
  migrate:up [<name>]                     Run the next or the specified migration that has not yet been run.
  migrate:rollback [options]              Rollback the last batch of migrations performed.
  migrate:down [<name>]                   Undo the last or the specified migration that was already run.
  migrate:currentVersion                  View the current version for the migration.
  migrate:list|migrate:status             List all migrations files with status.
  migrate:unlock                          Forcibly unlocks the migrations lock table.
  seed:make [options] <name>              Create a named seed file.
  seed:run [options]                      Run seed files.
  help [command]                  display help for command
Usage: cli [options] [command]

Options:
  -V, --version                   output the version number
  --debug                         Run with debugging.
  --knexfile [path]               Specify the knexfile path.
  --knexpath [path]               Specify the path to knex instance.
  --cwd [path]                    Specify the working directory.
  --client [name]                 Set DB client without a knexfile.
  --connection [address]          Set DB connection without a knexfile.
  --migrations-table-name [path]  Set migrations table name without a knexfile.
  --env [name]                    environment, default: process.env.NODE_ENV || development
  --esm                           Enable ESM interop.
  --specific [path]               Specify one seed file to execute.
  --timestamp-filename-prefix     Enable a timestamp prefix on name of generated seed files.
  -h, --help                      display help for command

Commands:
  init [options]                          Create a fresh knexfile.
  migrate:make [options] <name>           Create a named migration file.
  migrate:latest [options]                Run all migrations that have not yet been run.
  migrate:up [<name>]                     Run the next or the specified migration that has not yet been run.
  migrate:rollback [options]              Rollback the last batch of migrations performed.
  migrate:down [<name>]                   Undo the last or the specified migration that was already run.
  migrate:currentVersion                  View the current version for the migration.
  migrate:list|migrate:status             List all migrations files with status.
  migrate:unlock                          Forcibly unlocks the migrations lock table.
  seed:make [options] <name>              Create a named seed file.
  seed:run [options]                      Run seed files.
  help [command]                  display help for command
```

**Comment by wejesuss** [comment]
Hi @emilwihlander, I looked at the code and saw the following:
```js
if (!process.argv.slice(2).length) {
  commander.outputHelp();
}

commander.parse(process.argv);
```
I realized that using some command (like `help`) or option (like `--version`) the bug not occurs, the `if` will print the help in case of no command/option, but the problem is:
```js
commander.parse(process.argv);
```
This line also prints the help with no command/option, causing the bug.

I then commented on these lines
```js
// if (!process.argv.slice(2).length) {
  // commander.outputHelp();
// }
```
and seems to work for me.

**Comment by elhigu** [comment]
Commenting out that probably stops printing out help when client is called without any parameters, which is not desirable.

**Comment by emilwihlander** [comment]
@elhigu As mentioned by @wejesuss: `commander.parse(process.argv);` prints the help messages if no argument if given therefore making the help text printed twice. By removing the lines @wejesuss mention it prints the help text once for the following commands:
```
knex
knex --help
knex -h
```
I looked at the tests for the cli and it seems as though the library used to test it only support checking the beginning of the text printed. I, therefore, haven't found an easy way to create a test for repeated printing of the message. 

I will make a pull request fixing this.

**Comment by kibertoad** [comment]
Released in 0.21.9
